apollo-router/src/router.rs

```diff
@@ -606,8 +467,8 @@ where
             shutdown_receiver,
         );
 
-        let state_machine =
-            StateMachine::new(server_factory, Some(state_listener), self.router_factory);
+        let state_machine = StateMachine::new(server_factory, self.router_factory);
+        let ready = state_machine.listen_address.clone();
```

Comment:
```suggestion
        let listen_address = state_machine.listen_address.clone();
```

Reply:
Fixed in [1ce30b1](https://github.com/apollographql/router/pull/1227/commits/1ce30b17c04b57d20e8c57f4026197e63832d92b)